python/ray/tune/trial.py

```diff
@@ -497,6 +516,11 @@ def update_last_result(self, result, terminate=False):
         self.last_update_time = time.time()
         self.result_logger.on_result(self.last_result)
 
+        if not math.isnan(result["episode_reward_mean"]):
```

Comment:
also check first "episode_reward_mean" in result, I'm not sure each and every trainable has that one in each iteration, it would maybe be better not to break if that one is not inside, maybe just log a warning